servers/text_server.cpp

```diff
@@ -1120,6 +1121,29 @@ int TextServer::shaped_text_hit_test_position(RID p_shaped, real_t p_coords) con
 	return 0;
 }
 
+Vector2 TextServer::shaped_text_get_grapheme_bounds(RID p_shaped, int p_pos) const {
+	int v_size = shaped_text_get_glyph_count(p_shaped);
+	const Glyph *glyphs = shaped_text_get_glyphs(p_shaped);
+
+	real_t off = 0.0f;
+	for (int i = 0; i < v_size; i++) {
+		for (int k = 0; k < glyphs[i].repeat; k++) {
+			if ((glyphs[i].count > 0) && ((glyphs[i].index != 0) || ((glyphs[i].flags & GRAPHEME_IS_SPACE) == GRAPHEME_IS_SPACE))) {
+				if (glyphs[i].start <= p_pos && glyphs[i].end >= p_pos) {
+					real_t advance = 0.f;
+					for (int j = 0; j < glyphs[i].count; j++) {
+						advance += glyphs[i + j].advance;
+					}
+					return Vector2(off, off + advance);
+				}
+			}
+			off += glyphs[i].advance;
+		}
```

Comment:
Also, isn't it the same as `shaped_text_get_selection(rid, pos, pos + 1)` which can additionally handle parts of ligature.

Reply:
Yeah, I looked at `shaped_text_get_selection` first, but it seemed to not work properly at the end of the line because you had to supply it with the next index, which is non-existent. Unless I was missing something, of course. So I took the relevant parts and made a new targeted method. It doesn't actually need to handle anything else, as the index cannot be partially inside of a ligature. It's either a part of it, or it's not.